src/dmd/target.d

```diff
@@ -774,6 +776,60 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+
+
+    // Table of data for platformInfo
+    struct PlatformInfo
+    {
+        string key;
+        Expression function(ref const(Loc)) value;
+    }
+    enum platformInfo = [
+        PlatformInfo("cxxlib", (ref const(Loc) loc) => new StringExp(loc, cast(void*)global.params.mscrtlib, global.params.mscrtlib ? strlen(global.params.mscrtlib) : 0))
+    ];
+
+    /**
+     * Get targetInfo by key
+     * Params:
+     *  name = name of targetInfo to get
+     *  loc = location to use for error messages
+     * Returns:
+     *  Expression for the requested targetInfo
+     */
+    extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
+    {
+        switch (name.toDString)
+        {
+            static foreach (e; platformInfo)
```

Comment:
Sorry, `static foreach` won't be allowed in the compiler for at least another year.

If someone wants to backport the implementation of static foreach to C++ so it lands in GCC-9, then you'll reduce the time you have to wait.

Reply:
I thought you said that GDC and LDC have local forks of `target.d`? I put that magic in `target.d` with the specifically intent that it shouldn't affect GDC/LDC... is that wrong?

Reply:
Oh. I see.
Okay, I guess I'll expand the whole thing...